src/coreclr/src/vm/gccover.cpp

```diff
@@ -1411,6 +1411,32 @@ BOOL OnGcCoverageInterrupt(PCONTEXT regs)
         return TRUE;
     }
 
+    // If the thread is in preemptive mode then we must be in a
+    // PInvoke stub with an inlined frame, or be in a reverse PInvoke
+    // stub that's about to return.
+    //
+    // A PInvoke stub should should properly report GC refs if we
+    // trigger GC here. But a reverse PInvoke stub may over-report
+    // leading to spurious failures, as we would not normally report
+    // anything for this method.
+    //
+    // So, suppress GC for the reverse PInvoke stubs.
+    if (!pThread->PreemptiveGCDisabled())
+    {
+        _ASSERTE(pMD->IsILStub());
```

Comment:
```suggestion
        _ASSERTE(pMD->HasUnmanagedCallersOnlyAttribute());
```
The method does not have to be IL stub.

Reply:
Scratch that. It can be any method. PInvoke stub with an inlined frame can appear in any method.

Reply:
Good point. That was just some extra paranoia.